tools/darwin/Support/Codesign.command

```diff
@@ -2,66 +2,49 @@
 
-GEN_ENTITLEMENTS="$NATIVEPREFIX/bin/gen_entitlements.py"
 DARWIN_EMBEDDED_ENTITLEMENTS="$XBMC_DEPENDS/share/darwin_embedded_entitlements.xml"
 LDID="$NATIVEPREFIX/bin/ldid"
+TOPSHELF_BINARY="${BUILT_PRODUCTS_DIR}/${EXECUTABLE_FOLDER_PATH}/PlugIns/${TOPSHELF_EXTENSION_NAME}.appex/${TOPSHELF_EXTENSION_NAME}"
 
 if [ "${PLATFORM_NAME}" == "macosx" ]; then
   MACOS=1
 fi
 
-if [[ ! "$MACOS" && ! -f ${GEN_ENTITLEMENTS} ]]; then
-  echo "error: $GEN_ENTITLEMENTS not found. Codesign won't work."
-  exit -1
-fi
-
 if [ "$MACOS" ]; then
   CONTENTS_PATH="${CODESIGNING_FOLDER_PATH}/Contents"
 else
   CONTENTS_PATH="${CODESIGNING_FOLDER_PATH}"
 fi
 
-if [ ! "$MACOS" ]; then
-  # do fake sign - needed for iOS >=5.1 and tvOS >=10.2 jailbroken devices
-  # see http://www.saurik.com/id/8
-  "${LDID}" -S"${DARWIN_EMBEDDED_ENTITLEMENTS}" "${BUILT_PRODUCTS_DIR}/${EXECUTABLE_FOLDER_PATH}/${EXECUTABLE_NAME}"
-fi
-
-# pull the CFBundleIdentifier out of the built xxx.app
+# Pull the CFBundleIdentifier out of the built xxx.app
 BUNDLEID=$(/usr/libexec/PlistBuddy -c 'Print :CFBundleIdentifier' "${CONTENTS_PATH}/Info.plist")
-echo "CFBundleIdentifier is ${BUNDLEID}"
+echo "CFBundleIdentifier is '${BUNDLEID}'"
 
 # Prefer the expanded name, if available.
 CODE_SIGN_IDENTITY_FOR_ITEMS="${EXPANDED_CODE_SIGN_IDENTITY_NAME}"
 if [ "${CODE_SIGN_IDENTITY_FOR_ITEMS}" = "" ] ; then
   # Fall back to old behavior.
   CODE_SIGN_IDENTITY_FOR_ITEMS="${CODE_SIGN_IDENTITY}"
 fi
-echo "${CODE_SIGN_IDENTITY_FOR_ITEMS}"
-
-if [ ! "$MACOS" ]; then
-  ${GEN_ENTITLEMENTS} "${BUNDLEID}" "${BUILT_PRODUCTS_DIR}/${EXECUTABLE_FOLDER_PATH}/${EXECUTABLE_NAME}.xcent"
-  if [ -f "${CONTENTS_PATH}/embedded.mobileprovision" ]; then
-    rm -f "${CONTENTS_PATH}/embedded.mobileprovision"
-  fi
-fi
+echo "Code sign identity is '${CODE_SIGN_IDENTITY_FOR_ITEMS}'"
 
-# delete existing codesigning
-if [ -d "${CONTENTS_PATH}/_CodeSignature" ]; then
-  rm -r "${CONTENTS_PATH}/_CodeSignature"
-fi
-
-#if user has set a code_sign_identity different from iPhone Developer we do a real codesign (for deployment on non-jailbroken devices)
+# If user has set a code_sign_identity we do a real codesign (for deployment on non-jailbroken devices)
 if ! [ -z "${CODE_SIGN_IDENTITY_FOR_ITEMS}" ]; then
   if egrep -q --max-count=1 -e '^iPhone (Developer|Distribution): ' -e '^Apple (Development|Distribution): ' -e '^[[:xdigit:]]+$' -e '^Developer ID Application: ' <<<"${CODE_SIGN_IDENTITY_FOR_ITEMS}"; then
-    echo "Doing a full bundle sign using genuine identity ${CODE_SIGN_IDENTITY_FOR_ITEMS}"
+    echo "Doing a full bundle sign using genuine identity '${CODE_SIGN_IDENTITY_FOR_ITEMS}'"
+    
+    # Delete existing codesign and provisionong file
+    # (only with maxdepth=1 because we want to keep existing TopShelf binary signature)
+    find "${CONTENTS_PATH}" -maxdepth 1 -name "embedded.mobileprovision" -exec rm {} \;
+    find "${CONTENTS_PATH}" -maxdepth 1 -name "_CodeSignature" -exec rm -r {} \;
```

Comment:
I don't find this correct. These files should be deleted before deciding if we're signing or not.

Besides, the files reside in predefined locations, no need to use `find`.

Reply:
Why this is not correct? We care about these files only when we perform real signing, no? If we perform fake signing we don't care if these files are here or not?

I use `find` to handle the case where theses files do not exist.

Reply:
Imagine the following use case. You built a signed app, then changed project configuration to build a non-signed app. You already have app bundle with code signature: with your change existing code signature won't be deleted which is wrong.

Using `-f` flag on `rm` won't spit error message and return error code. No need to use `find`, it's not meant for that.

Reply:
Ok I understand what you said. For you it's wrong to packages  _CodeSignature and embedded.mobileprovision in a deb file that will be fake signed. I'm right?

I will replace find by rm -f and move these lines out of the real signing "if" :-)

Reply:
absolutely correct!